### ISSUE TITLE
Clarification on backward compatibility

### ISSUE THREAD

**Issue by heydonovan** [mozilla/sops#589]
Taken from the README:

>sops will remain backward compatible on the major version, meaning that all improvements brought to the 1.X and 2.X branches (current) will maintain the file format introduced in 1.0.

I'm not seeing these branch names, though I'm sure I've the gist of it.

We use 3.X right now, so as long as I'm using 3.X, all 3.X files should just work right? And when 4.X comes out, will 3.X versions work too? Or should we stick to 3.X only managing 3.X versions?

**Comment by ajvb** [comment]
@heydonovan Yeah, all `3.X` files will work with `3.X`. In regards to `4.X`, at this time we have no plans so I don't think we can safely set expectations. If there was a change in the file format, there would definitely be an easy upgrade path to `4.X`, but again we can't say at this time what that would look like.
